pandas/core/common.py

```diff
@@ -235,7 +236,16 @@ def asarray_tuplesafe(values: Iterable, dtype: NpDtype | None = None) -> ArrayLi
     if isinstance(values, list) and dtype in [np.object_, object]:
         return construct_1d_object_array_from_listlike(values)
 
-    result = np.asarray(values, dtype=dtype)
+    try:
+        with warnings.catch_warnings():
+            # Can remove warning filter once NumPy 1.24 is min version
+            warnings.simplefilter("ignore", np.VisibleDeprecationWarning)
+            result = np.asarray(values, dtype=dtype)
+    except ValueError:
+        # More performant than checking is_list_like over each element
```

Comment:
```suggestion
        # Using try/except since it's more performant than checking is_list_like over each element
```

No big difference, but I think it helps understand what's this about faster.